src/main/resources/python/code_block.py

```diff
@@ -1,25 +1,37 @@
 class Code_block:
-    def __init__(self, code, text_color=WHITE, text_weight=NORMAL, font="Times New Roman"):
+    def __init__(self, code, syntax_highlighting=True, syntax_highlighting_style="inkpot", text_color=WHITE, text_weight=NORMAL, font="Times New Roman", tab_spacing=2):
         group = VGroup()
-        fp = open("sample.re", "w")
 
-        for c in code:
-            for sc in c:
-                fp.write(sc + "\n")
+        if syntax_highlighting:
+            fp = open("sample.re", "w")
```

Comment:
What is this? Is this reading in a file?

Reply:
Yeah, basically we are using the syntax highlighting for a language called ReasonML because it was the most similar to our language. The Manim `Code` object only accepts a file of the same type as the language specified for syntax highlighting, so we create this `sample.re` ReasonML file every time

Reply:
Ah I see, can you make this a temp file then? Instead of in the root of the project

Reply:
We don't want to randomly spawn more files than we need to when the manim is running, it will end up in the users directory they compiled with

Reply:
https://docs.python.org/3/library/tempfile.html this should do it